api/api.go

```diff
@@ -112,3 +125,342 @@ func (api *API) statsHandler(ctx router.HTTPContext) {
 		ctx.UnsupportedAPIMethod()
 	}
 }
+
+// ============================== HELPER FUNCTIONS ============================== //
+
+func postEntityHandler(ctx router.HTTPContext, c db.Connector) {
+	// Decode request body into entity var
+	var entity db.Entity
+	err := json.NewDecoder(ctx.Request.Body).Decode(&entity)
+	if err != nil {
+		ctx.APIError(router.ErrInternal, "Can't decode request body (error: %s)", err)
+		return
+	}
+
+	ConvertFloat64ToInt(entity)
+
+	for _, v := range entity {
+		if !validValueType(v) {
+			ctx.APIError(router.ErrBadRequest, "Invalid Value Type: %v. Value type must be string or int", reflect.TypeOf(v))
+		}
+	}
+
+	entities := []db.Entity{entity}
+	ids, err := c.CreateEntities(entities)
+	if err != nil {
+		if _, ok := err.(db.ErrCreate); ok {
+			ctx.APIError(router.ErrInternal, "Error creating entity (error: %s)", err)
+			return
+		} else {
+			ctx.APIError(router.ErrInternal, "Uknown error when creating entity(error: %s)", err)
+			return
+		}
+	}
+
+	out, err := json.Marshal(ids[0])
+	if err != nil {
+		ctx.APIError(router.ErrInternal, "Can't encode response (error: %s)", err)
+		return
+	}
+
+	fmt.Fprintln(ctx.Response, string(out))
+}
+
+func getEntityHandler(ctx router.HTTPContext, c db.Connector) {
+	if len(ctx.Arguments) != 2 {
+		ctx.APIError(router.ErrMissingParam, "Missing param: id")
```

Comment:
What's the 1st arg? Is it possible we'd have 3 and so id isn't missing, we just have more than expected? I guess what strikes me is that len of some args does not seem directly or obviously related to a param missing. No big deal, I just want to make sure error messages are good/accurate.

Reply:
The first arg is the path (such as `/api/v1/entity`). Given how the route is defined for this particular handler, there's no possibility to have a 3rd.

```
api.Router.AddRoute(API_ROOT+"entity", api.entityHandler, "api-entity")
api.Router.AddRoute(API_ROOT+"entity/"+REQUEST_ID_PATTERN, api.entityHandler, "api-entity")
```